app/Services/Backup/Databases/MssqlDatabase.php

```
@@ -0,0 +1,224 @@
+            escapeshellarg($inputPath),
+            escapeshellarg($bacpacPath),
+            $sqlpackage,
+        ));
+    }
+
+    /**
+     * sqlpackage Import refuses to run if the target database already exists,
+     * so we drop it first. ALTER ... SET SINGLE_USER WITH ROLLBACK IMMEDIATE
+     * evicts any lingering sessions before the DROP.
+     */
+    public function prepareForRestore(string $schemaName, BackupLogger $logger, bool $forceDatabase = false): void
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$forceDatabase'. (undefined)

(UnusedFormalParameter)